internal/fs/buffer/writebuffer.go

```diff
@@ -0,0 +1,18 @@
+package buffer
+
+type WriteBuffer interface {
+	// Create creates a buffer of 2*size passed as a parameter.
+	Create(size int) WriteBuffer
+
+	// Write writes to the buffer.
+	Write(data []byte, offset int64) error
```

Comment:
It should also return number of bytes written.

Reply:
This PR is just for buffer creation. This will be taken care of in a separate PR.